src/lsstseries/analysis/structurefunction2.py

```diff
@@ -112,10 +121,12 @@ def calc_sf2(
             fluxes_2d = [fluxes[mask] for mask in id_masks]
             errors_2d = [errors[mask] for mask in id_masks]
 
-            res = _sf2_single(
-                times_2d, fluxes_2d, errors_2d, bins=bins, combine=combine, method=method, sthresh=sthresh
+            sf_method = SF_METHODS[argument_container.sf_method](
+                times_2d, fluxes_2d, errors_2d, argument_container
```

Comment:
Here's where we make use of the `SF_METHODS` dictionary, to create an instance of the structure function calculator object.